internal/cmdutils/resolve/resolve.go

```diff
@@ -0,0 +1,152 @@
+package resolve
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"code-intelligence.com/cifuzz/internal/cmdutils"
+	"code-intelligence.com/cifuzz/internal/config"
+	"code-intelligence.com/cifuzz/util/regexutil"
+)
+
+var cmakeFuzzTestFileNamePattern = regexp.MustCompile(`add_fuzz_test\((?P<fuzzTest>[a-zA-Z0-9_.+=,@~-]+)\s(?P<file>[a-zA-Z0-9_.+=,@~-]+)\)`)
+var bazelFuzzTestFileNamePattern = regexp.MustCompile(`cc_fuzz_test *\([^)]* ?name *= *['"](?P<fuzzTest>[a-zA-Z0-9_.+=,@~-]*)['"]\,\s*srcs\s*=*\s\[\s*['"](?P<file>[a-zA-Z0-9_.+=,@~-]*)['"]`)
+
+// resolve determines the corresponding fuzz test name to a given source file.
+// The path has to be relative to the project directory.
+func resolve(path, buildSystem, projectDir string) (string, error) {
+	errNoFuzzTest := errors.New("No fuzz test found for the given source file")
+
+	switch buildSystem {
+	case config.BuildSystemCMake:
+		cmakeLists, err := findAllCMakeLists(projectDir)
+		if err != nil {
+			return "", err
+		}
+
+		for _, list := range cmakeLists {
+			var bs []byte
+			bs, err = os.ReadFile(filepath.Join(projectDir, list))
+			if err != nil {
+				return "", err
+			}
+
+			if !strings.Contains(string(bs), "add_fuzz_test") {
+				continue
+			}
+
+			matches, found := regexutil.FindAllNamedGroupsMatches(cmakeFuzzTestFileNamePattern, string(bs))
+			if found {
+				for _, match := range matches {
+					if (filepath.IsAbs(path) && filepath.Join(projectDir, filepath.Dir(list), match["file"]) == path) ||
+						filepath.Join(filepath.Dir(list), match["file"]) == path {
+						return match["fuzzTest"], nil
+					}
+				}
+			}
```

Comment:
We don't need to check `found` because `matches` is nil when no matches are found:
```suggestion
			matches, _ := regexutil.FindAllNamedGroupsMatches(cmakeFuzzTestFileNamePattern, string(bs))
			for _, match := range matches {
				if (filepath.IsAbs(path) && filepath.Join(projectDir, filepath.Dir(list), match["file"]) == path) ||
					filepath.Join(filepath.Dir(list), match["file"]) == path {
					return match["fuzzTest"], nil
				}
			}
```